src/sql/src/pure/postgres.rs

```diff
@@ -155,24 +159,25 @@ pub(super) fn generate_text_columns(
     Ok(text_cols_dict)
 }
 
-pub(crate) fn generate_targeted_subsources<F>(
+pub(crate) fn generate_targeted_subsources(
     scx: &StatementContext,
+    source_name: Option<ResolvedItemName>,
     validated_requested_subsources: Vec<RequestedSubsource<'_, PostgresTableDesc>>,
     mut text_cols_dict: BTreeMap<u32, BTreeSet<String>>,
-    mut get_transient_subsource_id: F,
     publication_tables: &[PostgresTableDesc],
 ) -> Result<
     (
-        Vec<CreateSourceSubsource<Aug>>,
-        Vec<(GlobalId, CreateSubsourceStatement<Aug>)>,
+        Vec<CreateSubsourceStatement<Aug>>,
+        // These are the tables referenced by the subsources. We want this set
+        // of tables separately so we can retain only table definitions that are
+        // referenced by subsources. This helps avoid issues when generating PG
+        // source table casts.
+        Vec<PostgresTableDesc>,
```

Comment:
You are correct, however both callers of this fn immediately want to split the values, so that seems annoying. The places that they're used that assert that the vecs are of the same length are also very far away.